src/V3AstNodes.h

```diff
@@ -306,11 +306,14 @@ class AstClass : public AstNodeModule {
     AstClassPackage* m_packagep = nullptr;  // Class package this is under
     bool m_virtual = false;  // Virtual class
     bool m_extended = false;  // Is extension or extended by other classes
+    void declareRandomizeMethod();
     void insertCache(AstNode* nodep);
 
 public:
     AstClass(FileLine* fl, const string& name)
-        : ASTGEN_SUPER(fl, name) {}
+        : ASTGEN_SUPER(fl, name) {
+        declareRandomizeMethod();
```

Comment:
I thought of this, and I know that you can walk the tree towards the root (through `AstExtends`), but what about going the other way? Does Verilator already provide something for this or should I implement storing this information in the AST?

Reply:
You can theoretically iterate backwards but generally this isn't done because it's easy to cause O(n^2) behavior if done badly, and generally can be avoided by remembering stuff.

Yes, store it in the AstClass; i.e. in some stage before you need the info, e.g. V3LinkResolve you'd see the randc and set a bool stored inside the AstClass.  This won't be perfect, e.g. something might remove the need for the randc, but that's ok it's just a hint that it might be needed, and can be wrong (it really isn't needed but says it is).